internal/app/networkd/pkg/nic/nic.go

```
@@ -2,75 +2,323 @@
+		return err
+	}
+
+	// Set link MTU if we got a response
+	if err = n.setMTU(method.Link().Index, method.MTU()); err != nil {
+		log.Printf("failed to set mtu %d for %s: %v", method.MTU(), method.Link().Name, err)
+		return err
+	}
+
+	// Check to see if we need to configure the address
+	addrs, err := n.rtnlConn.Addrs(method.Link(), method.Family())
+	if err != nil {
+		return err
+	}
+
+	addressExists := false
+
+	for _, addr := range addrs {
+		if method.Address().String() == addr.String() {
+			addressExists = true
+			break
+		}
+	}
+
+	if !addressExists {
+		if err = n.rtnlConn.AddrAdd(method.Link(), method.Address()); err != nil {
+			switch err := err.(type) {
+			case *netlink.OpError:
+				// ignore the error if it's -EEXIST or -ESRCH
+				if !os.IsExist(err.Err) && err.Err != syscall.ESRCH {
+					log.Printf("failed to add address %+v to %s: %v", method.Address(), method.Link().Name, err)
+					return err
+				}
+			default:
+				log.Printf("failed to add address (already exists) %+v to %s: %v", method.Address(), method.Link().Name, err)
+			}
+		}
+	}
+
+	// Add any routes
+	for _, r := range method.Routes() {
+		if err = n.rtnlConn.RouteAddSrc(method.Link(), *r.Dest, method.Address(), r.Router); err != nil {
+			switch err := err.(type) {
+			case *netlink.OpError:
+				// ignore the error if it's -EEXIST or -ESRCH
+				if !os.IsExist(err.Err) && err.Err != syscall.ESRCH {
+					log.Printf("failed to add route %+v for %s: %v", r, method.Link().Name, err)
+					continue
+				}
+			default:
+				log.Printf("failed to add route (already exists) %+v for %s: %v", r, method.Link().Name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Reset removes addressing configuration from a given link.
+func (n *NetworkInterface) Reset() {
+	var (
+		err  error
+		nets []*net.IPNet
+	)
+
+	if nets, err = n.rtnlConn.Addrs(n.Link, 0); err != nil {
+		log.Printf("failed to retrieve addresses for %s: %v", n.Name, err)
+		return
```

I feel like we should return an error here.